src/track.c

```diff
@@ -453,4 +469,51 @@ void tracking_send_state()
 
 }
 
+/** Parse a string describing the tracking loop filter parameters into
+    the loop_params_stage structs. */
+static bool parse_loop_params(struct setting *s, const char *val)
+{
+  /** The string contains loop parameters for either one or two
+      stages.  If the second is omitted, we'll use the same parameters
+      as the first stage.*/
+
+  if (s->type->from_string(s->type->priv, s->addr, s->len, val)) {
+    for (int stage = 0; stage < 2; stage++) {
+      struct loop_params *l = &loop_params_stage[stage];
+
+      int n_chars_read = 0;
+      if (sscanf(val, "( %hhu ms , ( %f , %f , %f , %f ) , ( %f , %f , %f , %f ) ) , %n",
```

Comment:
Maybe pull these out to temporary structs first, and copy them over `loop_params_stage` after checking.
As it is now if the parsing fails or `coherent_ms` is invalid we could be left with trashed parameters in memory.